phases/codesign.go

```diff
@@ -2,133 +2,175 @@ package phases
+				secret: true,
+				action: func(answer string) *option {
+					result.Android.Password = answer
+					return nil
+				}}).run()
+
+			(&option{
+				title: "Enter key alias",
+				action: func(answer string) *option {
+					result.Android.Alias = answer
+					return nil
+				}}).run()
+
+			(&option{
+				title:  "Enter key password",
+				secret: true,
+				action: func(answer string) *option {
+					result.Android.KeyPassword = answer
+					return nil
+				}}).run()
+		}
+	}
 
-					cmd := command.New("bash").SetStderr(os.Stderr).SetStdin(os.Stdin).SetStdout(os.Stdout)
-					if err = cmd.Run(); err != nil {
-						err = fmt.Errorf("failed to run command: %s, error: %s", cmd.PrintableCommandArgs(), err)
-					}
+	return result, err
+}
 
-					var files []os.FileInfo
-					files, err = ioutil.ReadDir(codesignExportsDir)
-					if err != nil {
-						return nil
-					}
+func evniromentsToMap(envs []envmanModels.EnvironmentItemModel) (map[string]string, error) {
+	nameToValue := map[string]string{}
 
-					for _, file := range files {
-						switch ext := strings.HasSuffix; {
-						case ext(file.Name(), ".p12"):
-							result.CertificatePaths = append(result.CertificatePaths, filepath.Join(codesignExportsDir, file.Name()))
-						case ext(file.Name(), ".mobileprovision"):
-							result.ProfilePaths = append(result.ProfilePaths, filepath.Join(codesignExportsDir, file.Name()))
-						}
-					}
-				}
+	for _, env := range envs {
+		key, value, err := env.GetKeyValuePair()
+		if err != nil {
+			return nil, err
+		}
+		nameToValue[key] = value
+	}
 
-				if projectType == platformAndroid || projectType == platformBoth {
-					(&option{
-						title: "Enter key store path",
-						action: func(answer string) *option {
-							result.KeystorePath = answer
-							return nil
-						}}).run()
-
-					(&option{
-						title:  "Enter key store password",
-						secret: true,
-						action: func(answer string) *option {
-							result.Password = answer
-							return nil
-						}}).run()
-
-					(&option{
-						title: "Enter key alias",
-						action: func(answer string) *option {
-							result.Alias = answer
-							return nil
-						}}).run()
-
-					(&option{
-						title:  "Enter key password",
-						secret: true,
-						action: func(answer string) *option {
-							result.KeyPassword = answer
-							return nil
-						}}).run()
-				}
+	return nameToValue, nil
+}
 
-				return nil
-			}
-			return nil
-		}}).run()
+func iosCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (CodesignResultsIOS, error) {
+	var xcodeProjects []xcode.CommandModel
 
-	return result, err
+	appEnvToValue, err := evniromentsToMap(bitriseYML.App.Environments)
+	if err != nil {
+		return CodesignResultsIOS{}, err
+	}
+
+	projectPath, pathOk := appEnvToValue["BITRISE_PROJECT_PATH"]
+	scheme, schemeOk := appEnvToValue["BITRISE_SCHEME"]
+
+	if !(pathOk && schemeOk) {
+		return CodesignResultsIOS{}, fmt.Errorf("could not find Xcode project path")
```

Comment:
in case of custom/user config, these fields may are not exposed, what if in this case, the tool asks for this info?